src/syncerd/types.rs

```diff
@@ -316,29 +356,64 @@ pub struct TaskAborted {
     pub error: Option<String>,
 }
 
-#[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
-#[display(Debug)]
+#[derive(Clone, Debug, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
 pub struct HeightChanged {
     pub id: TaskId,
     pub block: Vec<u8>,
     pub height: u64,
 }
 
+impl fmt::Display for HeightChanged {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "HeightChanged(id: {}, block: {}, height: {})",
+            self.id,
+            hex::encode(&self.block),
+            self.height,
+        )
+    }
+}
+
 #[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
-#[display(Debug)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+pub enum Txid {
```

Comment:
I think is it beneficial to implement `From<monero::Hash>` and `From<bitcoin::Txid>` on this struct. It allows to simplify a bunch of wrapping in the codebase by `.into()`

Reply:
3d17edd05e092d244c53b3fcf80c96b4b0e44f81